src/utils/dotnetUtils.ts

```diff
@@ -6,112 +6,114 @@
+import request = require('request-promise');
+import { QuickPickOptions } from "vscode";
+import { IActionContext, IAzureQuickPickItem } from 'vscode-azureextensionui';
+import { ProjectRuntime } from '../constants';
+import { ext } from '../extensionVariables';
 import { localize } from "../localize";
+import { tryGetTemplateVersionSetting } from '../templates/TemplateData';
+import { functionRuntimeUtils } from '../utils/functionRuntimeUtils';
 import { cpUtils } from "./cpUtils";
 import * as fsUtil from './fs';
+import { cliFeedJsonResponse, getFeedRuntime, tryGetCliFeedJson } from './getCliFeedJson';
 
-export namespace dotnetUtils {
-    const projectPackageId: string = 'microsoft.azurefunctions.projecttemplates';
-    const functionsPackageId: string = 'azure.functions.templates';
-    const betaTemplateVersion: string = '2.0.0-beta-10167';
-    const v1TemplateVersion: string = '1.0.3.10168';
-
-    function getPackagesCsproj(version: string): string {
-        // tslint:disable:no-multiline-string
-        return `<?xml version="1.0" encoding="utf-8"?>
-<Project Sdk="Microsoft.NET.Sdk">
-  <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
-  </PropertyGroup>
-  <ItemGroup>
-    <PackageReference Include="${projectPackageId}" Version="${version}" />
-    <PackageReference Include="${functionsPackageId}" Version="${version}" />
-  </ItemGroup>
-</Project>
-`;
-    }
+const projectPackageId: string = 'microsoft.azurefunctions.projecttemplates';
+const functionsPackageId: string = 'azure.functions.templates';
+const templatesKey: string = 'cSharpFunctionsTemplates';
 
-    async function validateDotnetInstalled(): Promise<void> {
-        try {
-            await cpUtils.executeCommand(undefined, undefined, 'dotnet', '--version');
-        } catch (error) {
-            throw new Error(localize('azFunc.dotnetNotInstalled', 'You must have the .NET CLI installed to perform this operation.'));
-        }
+async function validateDotnetInstalled(): Promise<void> {
+    try {
+        await cpUtils.executeCommand(undefined, undefined, 'dotnet', '--version');
+    } catch (error) {
+        throw new Error(localize('dotnetNotInstalled', 'You must have the .NET CLI installed to perform this operation.'));
     }
+}
+
+async function downloadDotnetTemplates(url: string, tempFolder: string, packageId: string): Promise<string> {
+    const fullPath: string = path.join(tempFolder, `${packageId}.nupkg`);
+    return new Promise(async (resolve: (fullPath: string) => void, reject: (e: Error) => void): Promise<void> => {
+        const templateOptions: request.OptionsWithUri = {
+            method: 'GET',
+            uri: url
+        };
+
+        request(templateOptions, (err: Error) => {
+            // tslint:disable-next-line:strict-boolean-expressions
+            if (err) {
+                reject(err);
+            }
+        }).pipe(fse.createWriteStream(fullPath).on('finish', () => {
```

Comment:
Have you tested if there's an error?  I think with node you get an unhandled exception which won't be caught if you don't have an error listener....?

Reply:
Fixed